test/checks/undefined_object_test.rb

```diff
@@ -91,45 +118,150 @@ def test_does_not_report_on_forloops
     assert_offenses("", offenses)
   end
 
-  def test_does_not_report_on_render
+  def test_report_on_render_with_variable_from_parent_context
+    offenses = analyze_theme(
+      ThemeCheck::UndefinedObject.new,
+      "templates/index.liquid" => <<~END,
+        {% assign price = "$3.00" %}
+        {% render 'product' %}
+      END
+      "snippets/product.liquid" => <<~END,
+        {{ price }}
+      END
+    )
+    assert_offenses(<<~END, offenses)
+      Undefined object: missing argument `price` at templates/index.liquid:2
```

Comment:
Can you drop the `Undefined object: ` prefix:

```suggestion
      Missing argument `price` at templates/index.liquid:2
```